modules/aws-eks-self-managed-node-groups/main.tf

```
@@ -6,6 +6,7 @@ resource "aws_autoscaling_group" "self_managed_ng" {
   vpc_zone_identifier = length(local.self_managed_node_group["subnet_ids"]) == 0 ? (local.self_managed_node_group["subnet_type"] == "public" ? var.context.public_subnet_ids : var.context.private_subnet_ids) : local.self_managed_node_group["subnet_ids"]
 
   capacity_rebalance = local.self_managed_node_group["capacity_rebalance"]
+  target_group_arns  = local.self_managed_node_group["target_group_arns"]
```

Instead of defaulting to an empty list, we should default to `null` - for any users that are using the `aws_autoscaling_attachment` resource, if we default to an empty list this will cause conflicts for them whereas `null` should not (I believe).  With the proposed change below, you can remove the change to the `locals.tf` file

See more info in the `Note` section at the top of https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group

```suggestion
  target_group_arns  = try(local.self_managed_node_group["target_group_arns"], null)
```

---

Thanks, updated